V/VLConvert/build_tarballs.jl

```diff
@@ -0,0 +1,34 @@
+using BinaryBuilder
+
+name = "VLConvert"
+version = v"1.1.0"
+
+sources = [
+    GitSource("https://github.com/vega/vl-convert.git", "ec4940b42b54bda3d9828323cf0de0e3c71bdf39"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd ${WORKSPACE}/srcdir/vl-convert*/
+cargo build --release
+install -D -m 755 "target/${rust_target}/release/vl-convert${exeext}" "${bindir}/vl-convert${exeext}"
+"""
+
+platforms = supported_platforms()
+# Our Rust toolchain for i686 Windows is unusable
+filter!(p -> !Sys.iswindows(p) || arch(p) != "i686", platforms)
+
+# The products that we will ensure are always built
+products = [
+    ExecutableProduct("vl-convert", :vlconvert),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(
+    ARGS, name, version, sources, script, platforms, products, dependencies;
+    compilers=[:c, :rust], julia_compat="1.6",
```

Comment:
```suggestion
    compilers=[:c, :rust], julia_compat="1.6", lock_microarchitecture=false,
```

Reply:
I'm not sure what that does. But it doesn't seem to have fixed it. 
The error that I see now (and apparently that error was also there before this change in some of the architectures) is:
```
[13:13:03] error: failed to run custom build command for `v8 v0.79.2`
[13:13:03] 
[13:13:03] Caused by:
[13:13:03]   process didn't exit successfully: `/workspace/srcdir/vl-convert/target/release/build/v8-dc75fb1ccbc98427/build-script-build` (exit status: 101)
[13:13:03]   --- stdout
[13:13:03]   cargo:rerun-if-changed=.gn
[13:13:03]   cargo:rerun-if-changed=BUILD.gn
[13:13:03]   cargo:rerun-if-changed=src/binding.cc
[13:13:03]   cargo:rerun-if-env-changed=CCACHE
[13:13:03]   cargo:rerun-if-env-changed=CLANG_BASE_PATH
[13:13:03]   cargo:rerun-if-env-changed=CXXSTDLIB
[13:13:03]   cargo:rerun-if-env-changed=DENO_TRYBUILD
[13:13:03]   cargo:rerun-if-env-changed=DOCS_RS
[13:13:03]   cargo:rerun-if-env-changed=GN
[13:13:03]   cargo:rerun-if-env-changed=GN_ARGS
[13:13:03]   cargo:rerun-if-env-changed=HOST
[13:13:03]   cargo:rerun-if-env-changed=NINJA
[13:13:03]   cargo:rerun-if-env-changed=OUT_DIR
[13:13:03]   cargo:rerun-if-env-changed=RUSTY_V8_ARCHIVE
[13:13:03]   cargo:rerun-if-env-changed=RUSTY_V8_MIRROR
[13:13:03]   cargo:rerun-if-env-changed=SCCACHE
[13:13:03]   cargo:rerun-if-env-changed=V8_FORCE_DEBUG
[13:13:03]   cargo:rerun-if-env-changed=V8_FROM_SOURCE
[13:13:03]   cargo:rerun-if-env-changed=PYTHON
[13:13:03]   cargo:rerun-if-env-changed=DISABLE_CLANG
[13:13:03]   cargo:rerun-if-env-changed=EXTRA_GN_ARGS
[13:13:03]   cargo:rerun-if-env-changed=NO_PRINT_GN_ARGS
[13:13:03]   cargo:rustc-link-lib=static=rusty_v8
[13:13:03]   download lockfile: "/workspace/srcdir/vl-convert/target/release/build/lib_download.fslock"
[13:13:03]   static lib URL: https://github.com/denoland/rusty_v8/releases/download/v0.79.2/librusty_v8_release_x86_64-unknown-linux-musl.a
[13:13:03]   cargo:rustc-link-search=/workspace/srcdir/vl-convert/target/release/gn_out/obj
[13:13:03]   Downloading https://github.com/denoland/rusty_v8/releases/download/v0.79.2/librusty_v8_release_x86_64-unknown-linux-musl.a
[13:13:03]   Downloading https://github.com/denoland/rusty_v8/releases/download/v0.79.2/librusty_v8_release_x86_64-unknown-linux-musl.a...
[13:13:03]   HTTP Error 404: Not Found
[13:13:03]   Python downloader failed, trying with curl.
[13:13:03] 
[13:13:03]   --- stderr
[13:13:03]   Traceback (most recent call last):
[13:13:03]     File "/opt/x86_64-linux-musl/registry/src/index.crates.io-6f17d22bba15001f/v8-0.79.2/./tools/download_file.py", line 64, in <module>
[13:13:03]       sys.exit(main())
[13:13:03]     File "/opt/x86_64-linux-musl/registry/src/index.crates.io-6f17d22bba15001f/v8-0.79.2/./tools/download_file.py", line 59, in main
[13:13:03]       DownloadUrl(args.url, f)
[13:13:03]     File "/opt/x86_64-linux-musl/registry/src/index.crates.io-6f17d22bba15001f/v8-0.79.2/./tools/download_file.py", line 45, in DownloadUrl
[13:13:03]       raise e
[13:13:03]     File "/opt/x86_64-linux-musl/registry/src/index.crates.io-6f17d22bba15001f/v8-0.79.2/./tools/download_file.py", line 29, in DownloadUrl
[13:13:03]       response = urlopen(url)
[13:13:03]     File "/usr/lib/python3.9/urllib/request.py", line 214, in urlopen
[13:13:03]       return opener.open(url, data, timeout)
[13:13:03]     File "/usr/lib/python3.9/urllib/request.py", line 523, in open
[13:13:03]       response = meth(req, response)
[13:13:03]     File "/usr/lib/python3.9/urllib/request.py", line 632, in http_response
[13:13:03]       response = self.parent.error(
[13:13:03]     File "/usr/lib/python3.9/urllib/request.py", line 561, in error
[13:13:03]       return self._call_chain(*args)
[13:13:03]     File "/usr/lib/python3.9/urllib/request.py", line 494, in _call_chain
[13:13:03]       result = func(*args)
[13:13:03]     File "/usr/lib/python3.9/urllib/request.py", line 641, in http_error_default
[13:13:03]       raise HTTPError(req.full_url, code, msg, hdrs, fp)
[13:13:03]   urllib.error.HTTPError: HTTP Error 404: Not Found
[13:13:03]   thread 'main' panicked at /opt/x86_64-linux-musl/registry/src/index.crates.io-6f17d22bba15001f/v8-0.79.2/build.rs:431:3:
[13:13:03]   assertion failed: status.success()
[13:13:03]   note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[13:13:03] warning: build failed, waiting for other jobs to finish...
[13:13:07]  ---> cargo build --release
[13:13:07]  ---> cargo build --release
[13:13:07] Previous command exited with 101
[13:13:07] Child Process exited, exit code 101
ERROR: LoadError: Build for VLConvert on x86_64-linux-gnu did not complete successfully
```

again, not sure  why that fails for this particular recipe and not for other similar rust recipes :thinking:

Reply:
>  I'm not sure what that does.

It addressed https://github.com/JuliaPackaging/Yggdrasil/pull/7730#issuecomment-1831738053

> But it doesn't seem to have fixed it.

Well, it did fix the error you reported above.  Now you're showing ***a different*** error.  Someone calls it progress.

> again, not sure why that fails for this particular recipe and not for other similar rust recipes 🤔

Perhaps because only this package uses the package v8 as dependency?  I'm not familiar with Rust, but looking at the link which is reported not to exist (error 404) I can open the repository https://github.com/denoland/rusty_v8.  First of all I can verify that indeed the asset file does not exist, so the build system is clearly trying to download something which doesn't exist, nice.  Then I go to the README of the repository and read the section "[Binary build](https://github.com/denoland/rusty_v8/blob/fa11bcd769cddf4635da511e9e6b495f6ecb0454/README.md#binary-build)"

> V8 is very large and takes a long time to compile. Many users will prefer to use a prebuilt version of V8. We publish static libs for every version of rusty v8 on [Github](https://github.com/denoland/rusty_v8/releases).
>
> Binaries builds are turned on by default: `cargo` build will initiate a download from github to get the static lib. To disable this build using the `V8_FROM_SOURCE` environmental variable.
>
> When making changes to rusty_v8 itself, it should be tested by build from source. The CI always builds from source.

which seems to suggest a solution to your problem.

Reply:
You are totally right, that fixed the first error I reported. Apologies if my message sounded rude, I definitely appreciate every second of time you spend on this issue, and every comment/hint/tip/help is really welcomed and valued :heart:.

Thanks for the pointer to v8 :pray:. I'm a bit lost in this domain, but that seems to be pointing in the right direction. I'll have a closer look and see if I can come up with something.

Reply:
For some reason all builds try to download the non existing `librusty_v8_release_x86_64-unknown-linux-musl.a` asset. Even the build for other architectures like `aarch64-linux-gnu` with an asset that should exist (`librusty_v8_release_aarch64-unknown-linux-gnu.a`) are trying to download the `x86_64-unknown-linux-musl.a` asset :thinking:

Reply:
using `V8_FROM_SOURCE=1` didn't help :disappointed: